### ISSUE TITLE
Spanner Support

### ISSUE THREAD

**Issue by Omzig** [ironmansoftware/powershell-universal#3314]
### Summary of the new feature / enhancement

As a developer, I want to use spanner in gcp so that i can be cloud friendly. 
- https://cloud.google.com/spanner?hl=en

**Comment by Omzig** [comment]
@adamdriscoll, how hard would it get this into v5? Google SQL would be preferred.

**Comment by Omzig** [comment]
i think we just need to make sure we have the PostgresSQL Spanner driver and all the Postgres stuff should work:


https://www.cdata.com/kb/tech/googlespanner-jdbc-postgresql-fdw.rst
https://www.infoq.com/news/2022/07/google-cloud-spanner-postgresql/

**Comment by adamdriscoll** [comment]
This won't make it in v5. I don't want to introduce another DB at this time. If the PostgreSQL driver works, that'd be great. Spanner does have an EF provider but adding another database adds a lot of overhead.

**Comment by Omzig** [comment]
> This won't make it in v5. I don't want to introduce another DB at this time. If the PostgreSQL driver works, that'd be great. Spanner does have an EF provider but adding another database adds a lot of overhead.

can we add whatever the whateverSpanner.PostgreSQL plugin to psu v5?

**Comment by adamdriscoll** [comment]
Yeah. It would be possible to produce this plugin yourself. We can make sure that gets documented. We are looking at making a NuGet package for this kind of this. The other database providers are plugins already using the classes that will be part of the package.
